BUILDER_DEV.md

```diff
@@ -1,390 +1,285 @@
+```
+
+* Run the build-srv makefile (NOTE - if you receive an error about cargo not being found, make sure you have removed the "secure_path" section of /etc/sudoers)
+
+```
+$ sudo make build-srv
+```
+
+* Open up the builder-worker file
+
+```
+$ sudo vim /hab/svc/builder-worker
+```
+
+* Add your Github Auth Key:
+
+**/hab/svc/builder-worker**
+```
+auth_token = ""
+```
+
+Should become:
+
+**/hab/svc/builder-worker**
+```
 auth_token = "<your github token>"
-bldr_url = "http://localhost:9636"
-auto_publish = true
-airlock_enabled = false
 ```
 
+* Run this command (it may take awhile) to start all the Builder services
+
+```
+$ sudo -E make bldr-run
+```
+
+### UI Setup
+
+* Open a separate shell on your VM
+* Install npm
+
+```
+$ sudo apt-get install npm
+```
+
+* Cd into the builder-web component
 
-(Note: If you want your log files to persist across restarts of your development machine, replace `/tmp` with some other directory. It *must* exist and be writable before you start the job server).
+```
+$ cd path/to/your/habitat/repo/components/builder-web
+```
 
-Now, modify the `Procfile` (located in your hab repo in the `support` folder) to point the api, sessionsrv, jobsrv and worker services to the previously created config files, e.g.
+* Install all dependencies
 
 ```
-api: target/debug/bldr-api start --config /home/your_alias/habitat/config_api.toml
-sessionsrv: target/debug/bldr-sessionsrv start --config /home/your_alias/habitat/config_sessionsrv.toml
-worker: target/debug/bldr-worker start --config /home/your_alias/habitat/config_worker.toml
-jobsrv: target/debug/bldr-jobsrv start --config /home/your_alias/habitat/config_jobsrv.toml
+$ npm install
 ```
 
-## Build the Builder services for the first time
-1. Open a new terminal window.
-1. Run `make build-srv`
+* Create the habitat.conf.js file
 
-Note: this only needs to be done the first time to get the pieces in place. If you clean the project, make sure to unset the variables show below when you do a clean build, otherwise the build will look for packages on your local build service!
+```
+$ cp habitat.conf.sample.js habitat.conf.js
+```
 
-## Run the Builder services
-1. Open a new terminal window.
-1. Export the following environment variables:
+* Start the UI
 
 ```
-export HAB_AUTH_TOKEN=<your github token>
-export HAB_BLDR_URL=http://localhost:9636
-export HAB_ORIGIN=<your origin>
+$ sudo npm start
 ```
 
-Now run `sudo -E make bldr-run` from the root of your hab repo.
+* Open up a browser and navigate to http://localhost:3000/#/pkgs - you should see the Habitat UI running locally.
 
-The first time this command runs, it will create the required databases. Let it run for a while, and then re-start it if there are errors (this is normal for the first time setup).
+### Creating an Origin
 
-When `make bldr-run` can proceed with all the services up and running, you are ready to proceed to the next step.
+* Navigate to http://localhost:3000/#/pkgs
+* Log in
```

Comment:
Following the instructions, I cannot login here. It may be caused by the missing `.secrets/builder-github-app.pem` file though, see previous comment about that :)

Reply:
That is likely why, I will add a note to the documentation saying to expect info on being able to setup Github auth without a secret from the Habitat 1Password Vault soon.